### ISSUE TITLE
Policy Controller edge-22.2.1 fails w/error=failed to load private key

### ISSUE THREAD

**Issue by ron1** [linkerd/linkerd2#7963]
### What is the issue?

Helm-based HA, CNI edge-22.2.1 install fails with message:
```
Internal error occurred: failed calling webhook \"[linkerd-policy-validator.linkerd.io](http://linkerd-policy-validator.linkerd.io/)\": Post \"[https://linkerd-policy-validator.linkerd.svc:443/?timeout=10s](https://linkerd-policy-validator.linkerd.svc/?timeout=10s)\": EOF.
```

### How can it be reproduced?

Install Linkerd edge-22.2.1 via Helm in HA and CNI modes with policyValidator.externalSecret=true and policyValidator.injectCaFrom=linkerd/webhook-issuer-tls.

Notice linkerd-destination log errors reported below.

### Logs, error output, etc

linkerd_policy_controller log reports the following error:
```
ERROR admission:connection{xx.xx.xxx.xxx:40624}: linkerd_policy_controller: error=failed to load private key
```

### output of `linkerd check -o short`

```
Linkerd check results are good for edge-22.2.1
```

### Environment

Rancher RKE2 1.21.5+rke2r1
CentOS 7.9

### Possible solution

Note that reverting the following linkerd-control-plane helm chart image version to a known earlier release tag fixes the problem.
```
policyController.image.version=edge-22.1.1
```

**Comment by olix0r** [comment]
How are you generating TLS credentials? What types of keys are you creating?

**Comment by ron1** [comment]
Created using cert-manager 1.6.1 according to linkerd byoc docs. Just reverting image tag to edge-22.1.1 fixes the problem.

**Comment by olix0r** [comment]
This is most likely related to https://github.com/linkerd/linkerd2/commit/96131b5ee885d1ab1bc47b0afdb600a0282be4f0 -- we'll need to investigate how the new certificate loading logic differs from warp's implementation.

**Comment by ron1** [comment]
That makes sense. Is it reasonably safe to revert back to the 22.1.1 policy controller image as a temporary work-around?

**Comment by olix0r** [comment]
@ron1 Yeah, though the older policy-controller won't automatically pick up new certificate versions when cert-manager rotates them -- that's what the change was intended to address. So you may need to manually restart the pods when cert-manager changes credentials.

Another option is just to remove the web hook configuration for now. The web hook does some validation `Server` & `ServerAuthorization` resources as they are created, but your control plane will operate properly without that admission controller.

**Comment by ron1** [comment]
@olix0r So just delete ValidatingWebhookConfiguration `linkerd-policy-validator-webhook-config`, right?

**Comment by olix0r** [comment]
@ron1 Correct!

**Comment by adleong** [comment]
I've tracked this issue down to a problem in our docs: https://linkerd.io/2.11/tasks/automatically-rotating-webhook-tls-credentials/#issuing-certificates-and-writing-them-to-secrets

In the `Certificate/linkerd-policy-validator` resource, we specify the private key to be a PKCS8 encoded ECDSA key:

```yaml
  privateKey:
    algorithm: ECDSA
    encoding: PKCS8
```

However, the policy controller expects a PKCS1 encoded RSA key.  Changing this configuration fixes the issue:

```
  privateKey:
    algorithm: RSA
```

Thanks for finding this, @ron1.  I'll work on getting the official docs updated to reflect this.

**Comment by olix0r** [comment]
@adleong We should be able to support ECDSA keys without a problem (as we did previously)... I'll see if I can sort that out from the library code
